pkg/providers/cloudstack/cloudstack.go

```diff
@@ -255,6 +257,7 @@ func NewProvider(datacenterConfig *v1alpha1.CloudStackDatacenterConfig, machineC
 			etcdMachineSpec:             etcdMachineSpec,
 			now:                         now,
 		},
+		log:       logger.Get(),
```

Comment:
This is great. Could we inject this as an argument in the constructor?

Reply:
ideally we want to generate the logger instance in factory as dependency setup and inject it to create the provider instance. we do not have that in any of the providers right now. I think it's ok, we can follow up w a separate PR for the change.

Reply:
We can call `logger.Get()` in the factory, that's not an issue. If the need to use custom loggers in the factory come up, we just need to update the factory.

IMO we should try to inject all dependencies in the constructor. We haven't always done that in the past, specially with the logger. But since you are already doing the hardest part (change all log entries to use the field logger), I would vote to inject it from the constructor and make it complete.

https://github.com/aws/eks-anywhere/blob/main/docs/developer/best-practice.md#-dependencies
> Avoid constructing dependencies in-line. If you do construct a dependency in-line because it compliments the most common usage, provide a way to override it such as with a setter.

https://github.com/aws/eks-anywhere/blob/main/docs/developer/best-practice.md#loggers